autorest/sender.go

```diff
@@ -295,14 +295,22 @@ func doRetryForStatusCodesImpl(s Sender, r *http.Request, count429 bool, attempt
 			return resp, err
 		}
 		delayed := DelayWithRetryAfter(resp, r.Context().Done())
-		if !delayed && !DelayForBackoffWithCap(backoff, cap, attempt, r.Context().Done()) {
+		// enforce a 2 minute cap when not counting 429 if there isn't one.
```

Comment:
I feel like this comment is a bit confusing. If I understand what we're trying to say here, I would phrase it like: enforce a 2 minute cap between requests when 429 status codes are not going to be considered and when the cap is 0.

I know it's more verbose, but it was a bit confusing at first.

Reply:
updated comment